src/redocly/index.js

```diff
@@ -0,0 +1,161 @@
+import {
+  existsSync, readFileSync, writeFileSync, unlinkSync,
+} from 'fs';
+import yaml from 'js-yaml';
+import { resolve } from 'path';
+import { homedir } from 'os';
+
+import query from './query';
+
+export default class RedoclyClient {
+  constructor() {
+    const credentialsPath = resolve(homedir(), '.redocly.token.json');
+    let credentials = {};
+    if (existsSync(credentialsPath)) {
+      credentials = JSON.parse(readFileSync(credentialsPath, 'utf-8'));
+    }
+    const authDetails = RedoclyClient.authorize(credentials.token);
```

Comment:
I don't think we should check if token is authorized each time + definitely not in constructor.

I would move it to `getAuthorizationHeader`. Also why we check it if we don't show any warning/error ?